Import in Typescript environment with NodeJS?

When I try importing with
`import moize from "moize/mjs/index.mjs";`

I get the following error:
`error TS7016: Could not find a declaration file for module 'moize/mjs/index.mjs'`

It seems like the type file is only found if I `import moize from 'moize';` however the build from that import causes other issues in NodeJS as well.

What's the recommended way to import `moize` via ESM or to configure `tsconfig.json` for an ESM Node environment with Typescript?

--- a/es-to-mjs.js
+++ b/es-to-mjs.js
@@ -5,42 +5,56 @@ const path = require('path');
 
 const pkg = require('./package.json');
 
-const SOURCE = path.join(__dirname, pkg.module);
-const SOURCE_MAP = `${SOURCE}.map`;
-const DESTINATION = path.join(__dirname, 'mjs', 'index.mjs');
-const DESTINATION_MAP = `${DESTINATION}.map`;
+const BASE_PATH = __dirname;
+const SOURCE_ENTRY = path.join(BASE_PATH, pkg.module);
+const SOURCE_MAP = `${SOURCE_ENTRY}.map`;
+const SOURCE_TYPES = path.join(BASE_PATH, 'index.d.ts');
+const DESTINATION = 'mjs';
+const DESTINATION_ENTRY = path.join(BASE_PATH, DESTINATION, 'index.mjs');
+const DESTINATION_MAP = `${DESTINATION_ENTRY}.map`;
+const DESTINATION_TYPES = path.join(BASE_PATH, DESTINATION, 'index.d.mts');
 
-const getFileName = (filename) => {
-  const split = filename.split('/');
-
-  return split[split.length - 1];
-};
+function getFileName(filename) {
+    return filename.replace(`${BASE_PATH}/`, '');
+}
 
 try {
-  if (!fs.existsSync(path.join(__dirname, 'mjs'))) {
-    fs.mkdirSync(path.join(__dirname, 'mjs'));
-  }
+    if (!fs.existsSync(path.join(__dirname, 'mjs'))) {
+        fs.mkdirSync(path.join(__dirname, 'mjs'));
+    }
 
-  fs.copyFileSync(SOURCE, DESTINATION);
+    fs.copyFileSync(SOURCE_ENTRY, DESTINATION_ENTRY);
 
-  const contents = fs
-    .readFileSync(DESTINATION, { encoding: 'utf8' })
-    .replace('fast-equals', 'fast-equals/dist/fast-equals.mjs')
-    .replace('fast-stringify', 'fast-stringify/mjs/index.mjs')
-    .replace('micro-memoize', 'micro-memoize/mjs/index.mjs')
-    .replace(/\/\/# sourceMappingURL=(.*)/, (match, value) => {
-      return match.replace(value, 'index.mjs.map');
-    });
+    const contents = fs
+        .readFileSync(DESTINATION_ENTRY, { encoding: 'utf8' })
+        .replace('fast-equals', 'fast-equals/dist/fast-equals.mjs')
+        .replace('fast-stringify', 'fast-stringify/mjs/index.mjs')
+        .replace('micro-memoize', 'micro-memoize/mjs/index.mjs')
+        .replace(/\/\/# sourceMappingURL=(.*)/, (match, value) => {
+            return match.replace(value, 'index.mjs.map');
+        });
 
-  fs.writeFileSync(DESTINATION, contents, { encoding: 'utf8' });
+    fs.writeFileSync(DESTINATION_ENTRY, contents, { encoding: 'utf8' });
 
-  console.log(`Copied ${getFileName(SOURCE)} to ${getFileName(DESTINATION)}`);
+    console.log(
+        `Copied ${getFileName(SOURCE_ENTRY)} to ${getFileName(
+            DESTINATION_ENTRY
+        )}`
+    );
 
-  fs.copyFileSync(SOURCE_MAP, DESTINATION_MAP);
+    fs.copyFileSync(SOURCE_MAP, DESTINATION_MAP);
 
-  console.log(`Copied ${getFileName(SOURCE_MAP)} to ${getFileName(DESTINATION_MAP)}`);
+    console.log(`Copied ${SOURCE_MAP} to ${getFileName(DESTINATION_MAP)}`);
+
+    fs.copyFileSync(SOURCE_TYPES, DESTINATION_TYPES);
+
+    console.log(
+        `Copied ${getFileName(SOURCE_TYPES)} to ${getFileName(
+            DESTINATION_TYPES
+        )}`
+    );
 } catch (error) {
-  console.error(error);
+    console.error(error);
 
-  process.exit(1);
+    process.exit(1);
 }

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,6 +1,10 @@
 /* eslint-disable */
 
-import { MicroMemoize } from 'micro-memoize/src/types';
+import {
+    Cache as BaseCache,
+    Memoized as BaseMemoized,
+    Options as BaseOptions,
+} from 'micro-memoize';
 
 export type AnyFn = (...args: any[]) => any;
 export type Moizeable = AnyFn & Record<string, any>;
@@ -36,9 +40,9 @@ export type Key<Arg extends any = any> = Arg[];
 export type Value = any;
 
 export type Cache<MoizeableFn extends Moizeable = Moizeable> =
-    MicroMemoize.Cache<MoizeableFn>;
+    BaseCache<MoizeableFn>;
 export type MicroMemoizeOptions<MoizeableFn extends Moizeable = Moizeable> =
-    MicroMemoize.Options<MoizeableFn>;
+    BaseOptions<MoizeableFn>;
 
 export type Expiration = {
     expirationMethod: () => void;
@@ -103,7 +107,7 @@ export type StatsCache = {
 };
 
 export type Memoized<MoizeableFn extends Moizeable = Moizeable> =
-    MicroMemoize.Memoized<MoizeableFn>;
+    BaseMemoized<MoizeableFn>;
 
 export type Moized<
     MoizeableFn extends Moizeable = Moizeable,

--- a/package.json
+++ b/package.json
@@ -14,7 +14,7 @@
     },
     "dependencies": {
         "fast-equals": "^3.0.1",
-        "micro-memoize": "^4.0.11"
+        "micro-memoize": "^4.1.2"
     },
     "description": "Blazing fast memoization based on all parameters passed",
     "devDependencies": {

--- a/package.json
+++ b/package.json
@@ -127,5 +127,5 @@
     },
     "sideEffects": false,
     "types": "./index.d.ts",
-    "version": "6.1.5"
+    "version": "6.1.6-beta.0"
 }

